src/index.ts

```diff
@@ -300,7 +289,68 @@ export abstract class DetSysAction {
     });
   }
 
-  // Whether the
+  getTemporaryName(): string {
+    const tmpDir = process.env["RUNNER_TEMP"] || tmpdir();
+    return path.join(tmpDir, `${this.actionOptions.name}-${randomUUID()}`);
+  }
+
+  addFact(key: string, value: string | boolean): void {
+    this.facts[key] = value;
+  }
+
+  getDiagnosticsUrl(): URL | undefined {
+    return this.actionOptions.diagnosticsUrl;
+  }
+
+  getUniqueId(): string {
+    return (
+      this.identity.run_differentiator ||
+      process.env.RUNNER_TRACKING_ID ||
+      randomUUID()
+    );
+  }
+
+  getCorrelationHashes(): correlation.AnonymizedCorrelationHashes {
+    return this.identity;
+  }
+
+  recordEvent(eventName: string, context: Record<string, unknown> = {}): void {
+    this.events.push({
+      event_name: `${this.actionOptions.eventPrefix}${eventName}`,
+      context,
+      correlation: this.identity,
+      facts: this.facts,
+      timestamp: new Date(),
+      uuid: randomUUID(),
+    });
+  }
+
+  /**
+   * Fetches a file in `.xz` format from the URL determined by the `source-*`
+   * parameters (or `source-binary` if set). It then imports that file's
```

Comment:
Can we reword this to be more "general" (such that the `source-binary` parenthetical is unnecessary)? Maybe something like:

> Unpacks the closure specified by the `source-*` parameters, imports the contents into the Nix store, and returns the path..........

It may even be worth it to completely ignore the existence of `source-*` in this doc comment and document the behavior of them in `fetchArtifact` which actually implements the `source-*` handling (AFAIU)?